### ISSUE TITLE
Export Invoices to new currency

### ISSUE THREAD

**Issue by maltfield** [wpovernight/woocommerce-pdf-invoices-packing-slips#878]
This ticket is a feature request to add a way for this plugin to export an invoice in a different currency.

# Why?

My store is in USD, but I have some customers in the EU that like to pay with SEPA transfer.

Currently I'm downloading the invoice in this plugin, opening the PDF in Libreoffice Draw, and manually editing the values before sending the invoice to the customer. I would prefer if I could just have this plugin automatically update the numerical line items, update the currency symbol, etc of the invoice at the time I download it.

# How?

Currently my workflow is:

1. Download invoice from wordpress
2. Open invoice PDF in LibreOffice Draw
3. Get current mid-market rate for USD -> EUR https://www.xe.com/currencyconverter/convert/?From=USD&To=EUR
4. Find & Replace all `$` to `€`
5. Loop through every line item and multiply the numerical value in the PDF by the mid-market rate obtained in step [3], rounding to the nearest hundredth decimal value.
6. Calculate the new total price, and update the `Total` field's numerical value.
7. Update the invoice header to include our business bank account's IBAN number

Basically, I'd want ^ this process implemented in PHP, with the following constraints:

1. Fetching the mid-market currency rates should be done by the client, not the server (to support hardened servers where firewalls and `WP_HTTP_BLOCK_EXTERNAL` block the server from initiating requests)
2. Javascript shouldn't be required (eg use `<noscript>` to add a "Fetch Rate" button that populates an input field using HTML)

**Comment by maltfield** [comment]
Would the devs be open to a PR for this feature? If so, please point me to the correct files that you think would need to be edited, and do let me know if you have any ideas regarding UI or otherwise.

**Comment by maltfield** [comment]
I'm also curious if it might make more sense to achieve this with a plugin that allows me to change the currency of an order:

 * https://wordpress.org/plugins/woo-multi-currency/
 * https://wordpress.org/plugins/woocommerce-multilingual/

**Comment by maltfield** [comment]
See also:

* https://wordpress.org/support/topic/admin-only-mode-order-specific-currency-change-after-order-is-placed/
 * https://wordpress.org/support/topic/wp_http_block_external-support-for-automatic-exchange-rates/
 * https://wordpress.org/support/topic/support-for-automatic-exchange-rates-wp_http_block_external/

**Comment by alexmigf** [comment]
Hi @maltfield 

We are working on adding compatibility between our Professional extension and the Curcy multi-currency plugin. Could you please reach out to us via [email](mailto:support@wpovernight.com) so we can discuss this further?

**Comment by maltfield** [comment]
@alexmigf I could email you, but wouldn't it be better to have the conversation publicly here on GitHub, so others can read it and join the discussion?

**Comment by alexmigf** [comment]
@maltfield Thanks for your feedback! We're glad to see interest in multi-currency conversion. Currently, we’re planning to include this feature in the paid version of the plugin, as it wasn't part of our initial vision for the free version. However, we’ll discuss it internally and evaluate how we can enhance the free version in the future. Feel free to share any additional ideas here, as community input is always valuable!
